src/test/common/test_crc32c.cc

```diff
@@ -265,3 +265,28 @@ TEST(Crc32c, RangeNull) {
     ASSERT_EQ(crc, *check);
   }
 }
+
+TEST(Crc32c, zeros_performance) {
+  utime_t start;
+  utime_t end;
+
+  for (size_t scale=1; scale < 31; scale++)
+  {
+    size_t size = (1<<scale) + rand()%(1<<scale);
+    start = ceph_clock_now();
+    uint32_t crc_a = ceph_crc32c(111, nullptr, size);
+    end = ceph_clock_now();
+    std::cout << "regular  method. size=" << size << " time= " << (double)(end-start)
+        << " at " << (double)size/(1024*1024)/(double)(end-start) << " MB/sec" << std::endl;
+
+    start = ceph_clock_now();
+    uint32_t crc_b = ceph_crc32c_sctp(111, nullptr, size);
```

Comment:
Yes, quite likely.  As @kestrels mentioned in his original post the test was non-scientific and just a "sniff test".  @kestrels is not online today.  I've re-run his test on that machine 10x times and averaged the results.  Intel fastzero assembly is still ~1.5x faster, but again the test case is not the best. I agree we should get a good representative set of input CRCs and buffer lengths and run the tests.  My only point is the test should be run vs the Intel fastzero optimization vs the slow fall back it is currently using.

RocksDB is also doing CRC.  If it is doing CRC on zero buffers we should see if this impl beats the _mm_crc32_u32 SSE42 instruction Rocks is using and if so, get this impl into RocksDB if it impacts BlueStore performance.